src/ast/passes/codegen_llvm.cpp

```diff
@@ -466,8 +466,19 @@ void CodegenLLVM::visit(Builtin &builtin)
     expr_ = b_.CreateLoad(b_.GetType(builtin.type), dst);
     b_.CreateLifetimeEnd(dst);
   } else if (builtin.ident == "probe") {
-    builtin.probe_id = get_probe_id();
-    expr_ = b_.getInt64(builtin.probe_id);
+    auto probe_str = probefull_;
+    probe_str.resize(builtin.type.GetSize() - 1);
+    Constant *const_str = ConstantDataArray::getString(module_->getContext(),
+                                                       probe_str,
+                                                       true);
+    Value *buf = b_.CreateProbeStrAllocation(builtin.type,
+                                             "probe",
+                                             builtin.loc);
+    b_.CreateMemsetBPF(buf, b_.getInt8(0), builtin.type.GetSize());
```

Comment:
The zeroing is not necessary for stack allocations, right? Just for the scratch buffer.

Reply:
Right. We can check for that and make this memset conditional.